### PR TITLE
#86 material3 date picker

### LINKED ISSUE #86: Replace DatePicker with Material3 DatePicker
https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#DatePicker(androidx.compose.material3.DatePickerState,androidx.compose.ui.Modifier,androidx.compose.material3.DatePickerFormatter,kotlin.Function0,kotlin.Function0,kotlin.Boolean,androidx.compose.material3.DatePickerColors)

### PR DESCRIPTION
**Pull request by MJJacobs01**
Good morning,

Added the Material3 DatePicker as per issue 86, please let me know if you are happy or if I should change anything?

Dose-App-DatePicker-Screenshot_20231024_055138

Resolved #86

### CODE REVIEW

**Comment by Wottrich** [comment]
Hey @MJJacobs01! 

It's important merge your branch with main branch of project. I have made some changes on Dialog (#85) and your changes conflicts with this!

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt [hunk 1]
```diff
@@ -457,7 +454,7 @@ private fun canSelectMoreTimesOfDay(selectionCount: Int, numberOfDosage: Int): B
     return selectionCount < numberOfDosage
 }
 
-private fun showMaxSelectionSnackbar(
+private fun showMaxSelectionSnackBar(
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt:457]: Let's keep this `showMaxSelectionSnackbar`.

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt [hunk 2]
```diff
@@ -518,34 +515,29 @@ fun RecurrenceDropdownMenu(recurrence: (String) -> Unit) {
...
-    val datePickerDialog =
-        DatePickerDialog(context, { _: DatePicker, year: Int, month: Int, dayOfMonth: Int ->
-            val newDate = Calendar.getInstance()
-            newDate.set(year, month, dayOfMonth)
-            selectedDate = "${month.toMonthName()} $dayOfMonth, $year"
-            endDate(newDate.timeInMillis)
-        }, year, month, day)
+    var selectedDate by rememberSaveable {
+        mutableStateOf(
+            datePickerState.selectedDateMillis!!.dateInMillisToFormattedString()
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt:538]: Can we avoid using `!!` and safely unwrap it instead?

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                }
+            },
+            content = {
+                DatePicker(
+                    state = state,
+                    showModeToggle = false,
+                    headline = {
+                        Text(
+                            modifier = Modifier.padding(start = 16.dp),
+                            text = state.selectedDateMillis!!.dateInMillisToFormattedString()
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt:59]: Safely unwrap here, as well. :)

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -18,4 +18,6 @@
     <string name="hours">hours</string>
     <string name="minutes">minutes</string>
     <string name="take_dose_now">Take your dose now</string>
+    <string name="confirmButton">OK</string>
+    <string name="dismissButton">Cancel</string>
```
Comment (waseefakhtar) [inline app/src/main/res/values/strings.xml:22]: Let's name these:

```suggestion
    <string name="ok">OK</string>
    <string name="cancel">Cancel</string>
```

**Review by waseefakhtar — COMMENTED** [review]
Looks pretty good! I just left some comments about null-safety and conventions. 

I also notice a merge conflict. It'd be great to take care of that too. :)

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt (hunk 1 above)
Comment (MJJacobs01) [reply app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/AddMedicationRoute.kt:457]: I have resolved this one

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt (hunk 3 above)
Comment (MJJacobs01) [reply app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt:59]: I have unwrapped this one in new commit

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (MJJacobs01) [reply app/src/main/res/values/strings.xml:22]: I have changed the string names

File: app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt (hunk 3 above)
Comment (waseefakhtar) [reply app/src/main/java/com/waseefakhtar/doseapp/feature/addmedication/DatePickerDialog.kt:59]: I can't seem to see the new changes you pushed.

Could it be that you pushed it to your forked repo and not here?

**Pull request closed without merging**